File: 078. Subsets/Java/Solution.java
public class Solution {
    public List<List<Integer>> subsets(int[] nums) {
        List<List<Integer>> result = new ArrayList<>();
        for (int i = 0; i < 1 << nums.length; i++) {
            List<Integer> current = new ArrayList<>();
            for (int j = 0; j < nums.length; j++) {
                if ((i & 1 << j) != 0) {
                    current.add(nums[j]);
                }
            }
            result.add(current);
        }
        return result;
    }
}

File: 078. Subsets/C++/Solution.cpp
class Solution {
public:
    vector<vector<int>> subsets(vector<int> &nums) {
        vector<vector<int>> result;
        for (int i = 0; i < 1 << nums.size(); i++) {
            vector<int> current;
            for (int j = 0; j < nums.size(); j++) {
                if (i & 1 << j) {
                    current.push_back(nums[j]);
                }
            }
            result.push_back(current);
        }
        return result;
    }
};